support-models/src/main/scala/com/gu/support/redemptions/RedemptionData.scala

```diff
@@ -2,24 +2,23 @@ package com.gu.support.redemptions
 
 import cats.syntax.functor._
 import com.gu.support.encoding.Codec
-import com.gu.support.redemptions.redemptions.{CorporateAccountId, RedemptionCode}
 import io.circe.syntax._
 import io.circe.{Decoder, Encoder}
 
 sealed abstract class RedemptionData(val redemptionCode: RedemptionCode)
 
-case class CorporateRedemption(override val redemptionCode: RedemptionCode, corporateAccountId: CorporateAccountId) extends RedemptionData(redemptionCode)
+case class CorporateRedemption(override val redemptionCode: RedemptionCode) extends RedemptionData(redemptionCode)
 
 object RedemptionData {
-    import Codec.deriveCodec
-    implicit val corporateCodec: Codec[CorporateRedemption] = deriveCodec
+  import Codec.deriveCodec
+  implicit val corporateCodec: Codec[CorporateRedemption] = deriveCodec
 
-    implicit val encoder: Encoder[RedemptionData] = Encoder.instance {
-      case c: CorporateRedemption => c.asJson
-    }
+  implicit val encoder: Encoder[RedemptionData] = Encoder.instance {
+    case c: CorporateRedemption => c.asJson
+  }
 
-    implicit val decoder: Decoder[RedemptionData] =
-      List[Decoder[RedemptionData]](
-        Decoder[CorporateRedemption].widen
-      ).reduceLeft(_ or _)
+  implicit val decoder: Decoder[RedemptionData] =
+    List[Decoder[RedemptionData]](
+      Decoder[CorporateRedemption].widen
```

Comment:
will this be able to distinguish between CorporateRedemption and GiftRedemption if they both have the same field or one is a subset of the other?  A bit hypothetical since this code is just-in-case, but we will need to be aware.

Reply:
If they have exactly the same field then we would need a type descriminator to aid the serialisation, but as you mentioned above, at that point we may not need the subclasses any more